src/algebra/gcd_monoid/div.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2022 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+import algebra.gcd_monoid.finset
+import algebra.gcd_monoid.basic
+import ring_theory.int.basic
+import ring_theory.polynomial.content
+
+/-!
+# Basic results about setwise gcds on normalized gcd monoid with a division
+This file proves some basic results about `finset.gcd` on `ℤ`.
+
+## Main results
+
+* `finset.nat.gcd_is_unit_of_div_gcd`: given a nonempty finset `s` and a function `f` from `s` to
+  `ℕ`, if `d = s.gcd`, then the `gcd` of `(f i) / d` equals `1`.
+* `finset.int.gcd_is_unit_of_div_gcd`: given a nonempty finset `s` and a function `f` from `s` to
+  `ℤ`, if `d = s.gcd`, then the `gcd` of `(f i) / d` equals `1`.
+* `finset.polynomial.gcd_is_unit_of_div_gcd`: given a nonempty finset `s` and a function `f` from
+  `s` to `K[X]`, if `d = s.gcd`, then the `gcd` of `(f i) / d` equals `1`.
+
+## TODO
+Add a typeclass to state these results uniformly.
+
+-/
+
+namespace finset
+
+namespace nat
+
+/-- Given a nonempty finset `s` and a function `f` from `s` to `ℕ`, if `d = s.gcd`,
+then the `gcd` of `(f i) / d` is equal to `1`. -/
+theorem coprime_of_div_gcd {β : Type*} {f : β → ℕ} (s : finset β) {x : β} (hx : x ∈ s)
+  (hfz : f x ≠ 0) : s.gcd (λ b, f b / s.gcd f) = 1 :=
+begin
+  obtain ⟨g, he, hg⟩ := finset.extract_gcd f ⟨x, hx⟩,
+  refine (finset.gcd_congr rfl $ λ a ha, _).trans hg,
+  rw [he a ha, nat.mul_div_cancel_left],
+  exact nat.pos_of_ne_zero (mt finset.gcd_eq_zero_iff.1 (λ h, hfz $ h x hx)),
+end
+
+theorem gcd_eq_one_of_div_gcd_id (s : finset ℕ) {x : ℕ} (hx : x ∈ s) (hnz : x ≠ 0) :
+  s.gcd (λ b, b / (s.gcd id)) = 1 :=
+coprime_of_div_gcd s hx hnz
+
+end nat
+
+namespace int
+
+/-- Given a nonempty finset `s` and a function `f` from `s` to `ℤ`, if `d = s.gcd`,
+then the `gcd` of `(f i) / d` is equal to `1`. -/
+theorem coprime_of_div_gcd {β : Type*} {f : β → ℤ} (s : finset β) {x : β} (hx : x ∈ s)
```

Comment:
I wouldn't put `of` in these names, since `of` specifically goes before hypotheses.
```suggestion
theorem coprime_div_gcd {β : Type*} {f : β → ℤ} (s : finset β) {x : β} (hx : x ∈ s)
```